app/decorators/models/course_decorator.rb

```diff
@@ -106,6 +106,7 @@ def set_course_start_end_time_from_school
 
     self.start_at = course_start_time_from_school
     self.conclude_at = course_end_time_from_school
+    self
```

Comment:
you may need to save here since it's an `after_create`

Reply:
changed to `before_create`